src/FileService.Data/FileRepository.cs

```diff
@@ -23,34 +23,23 @@ public FileRepository(IDataProvider provider, IHttpContextAccessor httpContextAc
       _httpContextAccessor = httpContextAccessor;
     }
 
-    public async Task<bool> CreateAsync(List<DbFile> files)
+    public async Task<List<Guid>> CreateAsync(List<DbFile> files)
     {
-      if (files == null || !files.Any())
-      {
-        return false;
-      }
-
       _provider.Files.AddRange(files);
       await _provider.SaveAsync();
 
-      return true;
+      return files.Select(f => f.Id).Where(x => !_provider.Files.Select(f => f.Id).Contains(x)).ToList();
```

Comment:
if you added files through addRenge, then all files were added, no need to make select, remove it

Reply:
OK